eng/azure-pipelines.yml

```diff
@@ -9,10 +9,129 @@ pr:
+            name: $(DncEngInternalBuildPool)
+            demands: ImageOverride -equals build.ubuntu.2204.amd64
+        container:
+          image: $(imagename)
+        steps:
+        - script: |
+            git clean -ffdx
+            git reset --hard HEAD
+            git submodule update --init
+            git submodule foreach git clean -ffdx
+          displayName: 'Clean up working directory and check out submodules'
+
+        - bash: |
+            ./build.sh --ci --restore --build --pack $(archflag) --configuration $(_BuildConfig) $(_InternalBuildArgs)
+          displayName: 'Build and package'
+
+      ############ MACOS BUILD ############
+      - job: Build_macOS
+        displayName: macOS
+        timeoutInMinutes: 480
+        variables:
+        - _BuildConfig: Release
+        strategy:
+          matrix:
+            x64:
+              assetManifestOS: osx
+              assetManifestPlatform: x64
+              archflag: --arch x64
+            arm64:
+              assetManifestOS: osx
+              assetManifestPlatform: arm64
+              archflag: --arch arm64
+        pool:
+          vmImage: macos-11
+        steps:
+        - script: |
+            git clean -ffdx
+            git reset --hard HEAD
+            git submodule update --init
+            git submodule foreach git clean -ffdx
+          displayName: 'Clean up working directory and check out submodules'
+
+        - bash: |
+            ./build.sh --ci --restore --build --pack $(archflag) --configuration $(_BuildConfig) $(_InternalBuildArgs)
+          displayName: 'Build and package'
+
+      ############ WINDOWS BUILD ############
+      - job: Build_Windows
+        displayName: Windows
+        timeoutInMinutes: 480
+        variables:
+        - _BuildConfig: Release
+        strategy:
+          matrix:
+            x64:
+              assetManifestOS: win
+              assetManifestPlatform: x64
+              archflag: -arch x64
+            arm64:
+              assetManifestOS: win
+              assetManifestPlatform: arm64
+              archflag: -arch arm64
+        pool:
+          ${{ if eq(variables['System.TeamProject'], 'public') }}:
+            name: $(DncEngPublicBuildPool)
+            demands: ImageOverride -equals windows.vs2022.amd64.open
+          ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+            name: $(DncEngInternalBuildPool)
+            demands: ImageOverride -equals windows.vs2022.amd64
+        steps:
+        - script: |
+            git clean -ffdx
+            git reset --hard HEAD
+            git submodule update --init
+            git submodule foreach git clean -ffdx
+          displayName: 'Clean up working directory and check out submodules'
+
+        - powershell: |
+            (New-Object System.Net.WebClient).DownloadFile('https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.sfx.exe', 'msys2.exe')
```

Comment:
can you please file an issue with dnceng so that they add this and the packages to the windows.vs2022.amd64 image instead?

Reply:
https://github.com/dotnet/arcade/issues/12938

Reply:
tl;dr from that discussion is "it ain't getting fixed any time soon due to changes in available staff time", but it's slightly improved for now.